Hardware/CPU/AMD10CPU.cs

```diff
@@ -51,29 +54,43 @@ internal sealed class AMD10CPU : AMDCPU {
     private readonly uint miscellaneousControlAddress;
     private readonly ushort miscellaneousControlDeviceId;
 
+    private const uint cStatesIoPort = 0xCD6;
+    private readonly byte cStatesIoOffset;
+
+    private readonly bool isSVI2;
+
     private readonly FileStream temperatureStream;
 
     private readonly double timeStampCounterMultiplier;
     private readonly bool corePerformanceBoostSupport;
 
     public AMD10CPU(int processorIndex, CPUID[][] cpuid, ISettings settings)
-      : base(processorIndex, cpuid, settings) 
-    {            
+      : base(processorIndex, cpuid, settings)
+    {
       // AMD family 1Xh processors support only one temperature sensor
       coreTemperature = new Sensor(
         "Core" + (coreCount > 1 ? " #1 - #" + coreCount : ""), 0,
         SensorType.Temperature, this, new [] {
             new ParameterDescription("Offset [°C]", "Temperature offset.", 0)
           }, settings);
 
+      coreVoltage = new Sensor(

```

Comment:
Please use CoreString for Core numbers.

Reply:
I change the name to "CPU Core", CoreString(i) return a wrong name, te processer has only 1 sensor for all cores.

Reply:
If there are multiple cores you can use CoreString, otherwise just Core**s** I think . You have it now as 1 of ... but that's never used elsewhere and rather pointless as you can already see how many there are going by the numbers.